price-estimator/src/models/query.rs

```diff
@@ -0,0 +1,69 @@
+//! Module implementing parsing request query parameters.
+
+use anyhow::{bail, Error, Result};
+use core::models::BatchId;
+use serde::Deserialize;
+use std::convert::TryFrom;
+
+/// Common query parameters shared across all price estimation routes.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(try_from = "RawQuery")]
+pub struct QueryParameters {
+    /// The unit of the token amounts.
+    pub unit: Unit,
+    /// The maximum number of hops (i.e. maximum ring trade length) used by the
+    /// `pricegraph` search algorithm.
+    pub hops: Option<usize>,
+    /// The generation to load the orderbook at.
+    pub generation: Generation,
+}
+
+/// Units for token amounts.
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Unit {
+    /// Atoms, smallest unit for a token.
+    Atoms,
+    /// Base units, such that `1.0` is `10.pow(decimals)` atoms.
+    BaseUnits,
+}
+
+/// When to perform a price estimate.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Generation {
```

Comment:
nit: I'm not a huge fan of this name, but can see how it's hard to find a good one. How about `PointInTime` or `EstimationTime`?

Reply:
:rofl: I had no idea what to name this so I literally googled "time synonym" and found it under "a portion of time in history or characterized by particular events or circumstances.". I also am not a huge fan of this name. I think I like `EstimationTime` the most.